Source/DafnyTestGeneration/DafnyInfo.cs

```diff
@@ -63,37 +377,88 @@ private class DafnyInfoExtractor : BottomUpVisitor {
           Visit(classDecl);
         } else if (d is IndDatatypeDecl datatypeDecl) {
           Visit(datatypeDecl);
+        } else if (d is NewtypeDecl newTypeDecl) {
+          Visit(newTypeDecl);
+        } else if (d is SubsetTypeDecl subsetTypeDecl) {
+          Visit(subsetTypeDecl);
+        } else if (d is TypeSynonymDeclBase typeSynonymDecl) {
+          Visit(typeSynonymDecl);
+        }
+      }
+
+      private void VisitUserDefinedTypeDeclaration(string newTypeName,
+        Type baseType, Expression/*?*/ witness, List<TypeParameter> typeArgs) {
+        if (witness != null) {
+          info.witnessForType[newTypeName] = witness;
+          if (DafnyOptions.O.TestGenOptions.Verbose) {
+            Console.Out.WriteLine($"// Values of type {newTypeName} will be " +
+                                  $"assigned the default value of " +
+                                  $"{Printer.ExprToString(info.witnessForType[newTypeName])}");
+          }
+        } else if (DafnyOptions.O.TestGenOptions.Verbose) {
+          Console.Out.WriteLine($"// Warning: Values of type {newTypeName} " +
```

Comment:
This sounds unsound to me, and I don't think just printing a warning is strong enough. The previous compromise was that a user could use the synthesized code even if the exit code was non-zero, so that it was a "you're voiding the warranty" kind of situation. That's the only way I'm okay with this.

Reply:
Got it, thank you for catching this - will redo.

Reply:
Done - Dafny will throw an error and eventually exit with a non-zero exit code (but will not stop generating tests so the user can decide what to do with the result)